### PR TITLE
add return reminders

### PR DESCRIPTION
**Pull request by skjerns**
I've added a script that runs every day at 7 am and reminds people of their items.

I tested it, however, it can always produce problems when out in the wild, so feel free to check if everything works as it should

### CODE REVIEW

**Comment by skjerns** [comment]
It seems to work! I received a mail for my return today :)

I set the workflow to run only when it's 7am (whichever timezone the github servers are?!), and not on push. Else we trigger it on each pull request

**Comment by skjerns** [comment]
@daniel17903 is this now already executed daily, even though it is in a branch?

File: ReturnReminderManager/mail_client.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    def _authenticate(self):
+        user = os.environ.get("MAIL_USER")
+        password = os.environ.get("MAIL_PASSWORD")
+        server = os.environ.get("MAIL_SERVER")
+        logging.debug(f'logging in to SMTP: {server} with mail {user}')
+        server = smtplib.SMTP(server, timeout=15)
+        server.starttls()
+        server.login(user, password)
+        self.server = server
+        self.leihlokal_mail = base64.b64decode(b'bGVpaGxva2FsQGJ1ZXJnZXJzdGlmdHVuZy1rYXJsc3J1aGUuZGU=').decode()
```
Comment (daniel17903) [inline ReturnReminderManager/mail_client.py:31]: What is this for? If you don‘t want the mail to be visible you could add it as a secret.

File: ReturnReminderManager/main.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  return bool(re.search(r"^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$", email))
+
+
+if __name__ == '__main__':
+
+    leihlokal = LeihLokal()
+    mail_client = MailClient()
+
+    # not sure how well this plays for the github timezone?
+    today = datetime.now().date()
```
Comment (daniel17903) [inline ReturnReminderManager/main.py:75]:
```suggestion
    import pytz
    today = datetime.now(pytz.timezone('Europe/Berlin')).date()
```

File: ReturnReminderManager/main.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    rentals_due_today = leihlokal.filter_rentals(lambda x: x.to_return_on==today)
```
Comment (daniel17903) [inline ReturnReminderManager/main.py:76]: Did you test if this works? I think the timestamps in the database always are at midnight.

**Comment by daniel17903** [comment]
Looks good 👍

**Review by daniel17903 — APPROVED** [review]

File: ReturnReminderManager/mail_client.py (hunk 1 above)
Comment (skjerns) [reply ReturnReminderManager/mail_client.py:31]: I just copied it from https://github.com/leih-lokal/scripts/blob/master/WPAppointmentManager/mail_client.py

The mail is not secret, but this way we protect ourselves a bit from spam I guess

File: ReturnReminderManager/main.py (hunk 3 above)
Comment (skjerns) [reply ReturnReminderManager/main.py:77]: Okay, so if they are at midnight and the script runs at 7am (github time?), this should work?

I'll test it in master and see if it sends them at the right time.

**Comment by skjerns** [comment]
great! :)

I changed the reminder send date to the day before return (as the voting has turned out in the signal group)

**Comment by skjerns** [comment]
So I checked, GitHub actions is running at UTC :) that should be fine then to run it at noon the day before the return

```
Local: 2021-07-06 07:44:11.479716
Berlin: 2021-07-06 09:44:11.479694+02:00
```

**Pull request merged**